### ISSUE TITLE
apt-get update should be called if apt-get install fails

### ISSUE THREAD

**Issue by koral--** [bitrise-io/bitrise#573]
One of the our bitrise [steps](https://github.com/DroidsOnRoids/bitrise-step-vpnc-connect/blob/master/step.yml) has the following dependency in `step.yml`:
```
deps:
  apt_get:
  - name: vpnc
```
At the time of writing installation of `iproute` which is a dependency of `vpnc` fails due to outdated URL:
```INFO[10:25:40] Start installing (vpnc) with apt-get         
(vpnc) isn't installed, installing...
sudo apt-get -y install vpnc failed -- out: (Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  iproute vpnc-scripts
Suggested packages:
  resolvconf dnsmasq openssh-server
The following NEW packages will be installed:
  iproute vpnc vpnc-scripts
0 upgraded, 3 newly installed, 0 to remove and 2 not upgraded.
Need to get 90.7 kB of archives.
After this operation, 294 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu xenial/universe amd64 vpnc-scripts all 0.1~git20150318-1 [12.3 kB]
Get:2 http://archive.ubuntu.com/ubuntu xenial/universe amd64 vpnc amd64 0.5.3r550-2build1 [76.0 kB]
Err:3 http://archive.ubuntu.com/ubuntu xenial-updates/main i386 iproute all 1:4.3.0-1ubuntu3.16.04.2
  404  Not Found [IP: 91.189.88.162 80]
Err:3 http://archive.ubuntu.com/ubuntu xenial-updates/main i386 iproute all 1:4.3.0-1ubuntu3.16.04.2
  404  Not Found [IP: 91.189.88.162 80]
Fetched 88.3 kB in 0s (200 kB/s)
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/i/iproute2/iproute_4.3.0-1ubuntu3.16.04.2_all.deb  404  Not Found [IP: 91.189.88.162 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?) err: (exit status 100)
INFO[10:25:46] Failed to install (vpnc) with apt-get        
WARN[10:25:46] Installing Step dependency failed, retrying ... 
INFO[10:25:46] Start installing (vpnc) with apt-get         
(vpnc) isn't installed, installing...
sudo apt-get -y install vpnc failed -- out: (Reading package lists...
```

CLI retries `sudo apt-get -y install vpnc` but it is pointless, subsequent attempts will also fail. However invoking `apt-get update` resolves the issue. `--fix-missing` parameter from the suggestion does not help.

**Comment by viktorbenei** [comment]
Thanks for reporting @koral-- !

We already had a related issue report (https://github.com/bitrise-io/bitrise/issues/443), but that was classified as enhancement back then.

Updated our internal card for this issue to point to this new github issue instead.

Solution in the meantime: run an `apt-get update` in a Script step (can be the very first step in the workflow).

We'll try to fix this in the CLI ASAP!

**Comment by kdobmayer** [comment]
Hi @koral-- !

Thanks for reporting. The issue is fixed in PR #584 . The new CLI version will be available from 13 March.
